### ISSUE TITLE
Angular.io site error with Safari on ipad ios 10.3.3

### ISSUE THREAD

**Issue by dgodard** [angular/angular#30928]
# 📚 Docs or angular.io bug report

### Description

Angular.io site error with Safari on ipad ios 10.3.3

The banner is the only element displayed.

### What's the affected URL?**
https://angular.io

## 🔥 Exception or Error
<pre><code>
[Error] Error: Zone.js has detected that ZoneAwarePromise `(window|global).Promise` has been overwritten.
Most likely cause is that a Promise polyfill has been loaded after Zone.js (Polyfilling Promise api is not necessary when zone.js is loaded. If you must load one, do so before loading zone.js.)
	assertZonePatched (polyfills-es5.2ca6f61437681bec2c52.js:1:1826)
	t (main-es5.7ac7814adf9fabaaadec.js:1:93725)
	bootstrapModuleFactory (main-es5.7ac7814adf9fabaaadec.js:1:100401)
	zUnb (main-es5.7ac7814adf9fabaaadec.js:1:503924)
	u (runtime-es2015.917e94f948b97825b571.js:1:512)
	(fonction anonyme) (main-es5.7ac7814adf9fabaaadec.js:1:2705)
	u (runtime-es2015.917e94f948b97825b571.js:1:512)
	t (runtime-es2015.917e94f948b97825b571.js:1:378)
	(fonction anonyme) (runtime-es2015.917e94f948b97825b571.js:1:2436)
	Code du module (runtime-es2015.917e94f948b97825b571.js:1:2455)
	evaluate
	moduleEvaluation
	(fonction anonyme)
	promiseReactionJob
[Error] ERROR
Error: [DocViewer] Error preparing document 'index': a@https://angular.io/runtime-es5.0fc1ca047c2e4621cbb7.js:1:1298 A@https://angular.io/polyfills-es5.2ca6f61437681bec2c52.js:1:18876 onInvokeTask@https://angular.io/main-es5.7ac7814adf9fabaaadec.js:1:94168 runTask@https://angular.io/polyfills-es5.2ca6f61437681bec2c52.js:1:3839 invokeTask@https://angular.io/polyfills-es5.2ca6f61437681bec2c52.js:1:9737 m@https://angular.io/polyfills-es5.2ca6f61437681bec2c52.js:1:22942 b@https://angular.io/polyfills-es5.2ca6f61437681bec2c52.js:1:23167
(fonction anonyme) — doc-viewer.component.ts:146
error — catchError.js.pre-build-optimizer.js:34
_error — tap.js.pre-build-optimizer.js:61
error — Subscriber.js.pre-build-optimizer.js:59
_error — Subscriber.js.pre-build-optimizer.js:79
error — Subscriber.js.pre-build-optimizer.js:59
_error — tap.js.pre-build-optimizer.js:61
error — Subscriber.js.pre-build-optimizer.js:59
notifyError — OuterSubscriber.js.pre-build-optimizer.js:13
_error — InnerSubscriber.js.pre-build-optimizer.js:18
error — Subscriber.js.pre-build-optimizer.js:59
c — es.promise.js.pre-build-optimizer.js:97
(fonction anonyme) — es.promise.js.pre-build-optimizer.js:114
r — microtask.js.pre-build-optimizer.js:25
onInvoke — core.js.pre-build-optimizer.js:24319
run — zone.js.pre-build-optimizer.js:150
(fonction anonyme) — zone.js.pre-build-optimizer.js:910
onInvokeTask — core.js.pre-build-optimizer.js:24310
runTask — zone.js.pre-build-optimizer.js:195
g — zone.js.pre-build-optimizer.js:601
invokeTask — zone.js.pre-build-optimizer.js:502
m — zone.js.pre-build-optimizer.js:1693
b — zone.js.pre-build-optimizer.js:1719
	be (main-es5.7ac7814adf9fabaaadec.js:1:30086)
	handleError (main-es5.7ac7814adf9fabaaadec.js:1:30293)
	handleError (main-es5.7ac7814adf9fabaaadec.js:1:500311)
	error (main-es5.7ac7814adf9fabaaadec.js:1:397196)
	(fonction anonyme) (main-es5.7ac7814adf9fabaaadec.js:1:466814)
	error (main-es5.7ac7814adf9fabaaadec.js:1:13712)
	_error (main-es5.7ac7814adf9fabaaadec.js:1:405909)
	error (main-es5.7ac7814adf9fabaaadec.js:1:143212)
	_error (main-es5.7ac7814adf9fabaaadec.js:1:143517)
	error (main-es5.7ac7814adf9fabaaadec.js:1:143212)
	_error (main-es5.7ac7814adf9fabaaadec.js:1:405909)
	error (main-es5.7ac7814adf9fabaaadec.js:1:143212)
	notifyError (main-es5.7ac7814adf9fabaaadec.js:1:178999)
	_error (main-es5.7ac7814adf9fabaaadec.js:1:376589)
	error (main-es5.7ac7814adf9fabaaadec.js:1:143212)
	c (polyfills-es5.2ca6f61437681bec2c52.js:1:103819)
	(fonction anonyme) (polyfills-es5.2ca6f61437681bec2c52.js:1:103964)
	r (polyfills-es5.2ca6f61437681bec2c52.js:1:47128)
	onInvoke (main-es5.7ac7814adf9fabaaadec.js:1:94253)
	run (polyfills-es5.2ca6f61437681bec2c52.js:1:3173)
	(fonction anonyme) (polyfills-es5.2ca6f61437681bec2c52.js:1:15146)
	onInvokeTask (main-es5.7ac7814adf9fabaaadec.js:1:94169)
	runTask (polyfills-es5.2ca6f61437681bec2c52.js:1:3840)
	g (polyfills-es5.2ca6f61437681bec2c52.js:1:10922)
	invokeTask (polyfills-es5.2ca6f61437681bec2c52.js:1:9761)
	m (polyfills-es5.2ca6f61437681bec2c52.js:1:22943)
	b (polyfills-es5.2ca6f61437681bec2c52.js:1:23168)

</code></pre>

### Browser info
Safari on Ipad 4 with ios 10.3.3

**Comment by TomySmit** [comment]
It has happened coz safari 10.3 loads all js (module and nonmodule). I fixed that reset for tsconfig and routing file/

**Comment by alan-agius4** [comment]
Blocked by https://github.com/angular/angular-cli/issues/14680

**Comment by gkalpak** [comment]
Should have been fixed by #31137.

**Comment by gkalpak** [comment]
Oh, right 😇 Thx for pointing it out.

I was confused, because you said this was blocked on angular/angular-cli#14680 and that was fixed by angular/angular-cli#14716, which was included in cli@8.1.0-beta.0, I thought #31137 (which upgraded angular.io to cli@8.1.0-beta.2) would fix it. But I forgot that Safari 10 is indeed not a supported browser version for angular.io (we target the last two major versions (so currently Safari 11 and 12).

In any case, this issue can be closed :grin:

**Comment by alan-agius4** [comment]
It was blocked on that PR, if indeed we wanted AIO to support the mentioned
browsers 😊

**Comment by Ks89** [comment]
I found exactly the same issue on a personal application with angular-* 8.0.x. It's happening only on Safari iOS10.

So, the solution is to wait for 8.1.0?

At the moment, I replaced "target: es2015" with "target: es5" in my tsconfig.json and it worked.

thanks

**Comment by alan-agius4** [comment]
@Ks89, no need to wait for 8.1.0. This fix is present in version `8.0.4` as well.

**Comment by Ks89** [comment]
@alan-agius4 it's not working, still the same issue with all packages updated and angular-cli 8.0.4:

Angular CLI: 8.0.4
Node: 10.15.1
OS: darwin x64
Angular:
...

Package                      Version
------------------------------------------------------
@angular-devkit/architect    0.800.4
@angular-devkit/core         8.0.4
@angular-devkit/schematics   8.0.4
@schematics/angular          8.0.4
@schematics/update           0.800.4
rxjs                         6.4.0

**Comment by alan-agius4** [comment]
Did you specify that you want to support the above mentioned browsers in browserslist?

**Comment by Ks89** [comment]
I'm using this browserlist:

```
> 0.5%
last 2 versions
Firefox ESR
not dead
IE 11
Firefox 67
Safari 10
Safari 11
ios_saf 10
ios_saf 11
```

npx browserslist shows:

```
and_chr 74
and_ff 66
and_qq 1.2
and_uc 11.8
android 67
android 4.4.3-4.4.4
baidu 7.12
chrome 74
chrome 73
chrome 72
chrome 49
edge 18
edge 17
firefox 67
firefox 66
firefox 65
firefox 60
ie 11
ie_mob 11
ios_saf 12.2
ios_saf 12.0-12.1
ios_saf 11.3-11.4
ios_saf 11.0-11.2
ios_saf 10.0-10.2
kaios 2.5
op_mini all
op_mob 46
opera 58
opera 57
safari 12.1
safari 12
safari 11
safari 10
samsung 9.2
samsung 8.2
samsung 4
```

This application doesn't start, but I also tried with another one of my company and it starts, but when I click on a button to navigate to a lazy loaded module I receive this error: 

```
[Error] ERROR
Error: Uncaught (in promise): Error: Runtime compiler is not loaded Co@https://my-company-webapp/en/main-es2015.bcd08324eaf7280a7df8.js:1:81813 https://my-company-webapp/en/main-es2015.bcd08324eaf7280a7df8.js:1:276928 _tryNext@https://my-company-webapp/en/main-es2015.bcd08324eaf7280a7df8.js:1:3172 _next@https://my-company-webapp/en/main-es2015.bcd08324eaf7280a7df8.js:1:3091 next@https://my-company-webapp/en/main-es2015.bcd08324eaf7280a7df8.js:1:5150 https://my-company-webapp/en/main-es2015.bcd08324eaf7280a7df8.js:1:199721 onInvoke@https://my-company-webapp/en/main-es2015.bcd08324eaf7280a7df8.js:1:83386 run@https://my-company-webapp/en/polyfills-es5.5257e1a0419cc7d43246.js:1:4476 https://my-company-webapp/en/polyfills-es5.5257e1a0419cc7d43246.js:1:16449 onInvokeTask@https://my-company-webapp/en/main-es2015.bcd08324eaf7280a7df8.js:1:83308 runTask@https://my-company-webapp/en/polyfills-es5.5257e1a0419cc7d43246.js:1:5143 g@https://my-company-webapp/en/polyfills-es5.5257e1a0419cc7d43246.js:1:12225 invokeTask@https://my-company-webapp/en/polyfills-es5.5257e1a0419cc7d43246.js:1:11064 b@https://my-company-webapp/en/polyfills-es5.5257e1a0419cc7d43246.js:1:24246 m@https://my-company-webapp/en/polyfills-es5.5257e1a0419cc7d43246.js:1:24471 — polyfills-es5.5257e1a0419cc7d43246.js:1:15705
	Ie (main-es2015.bcd08324eaf7280a7df8.js:1:25298)
	handleError (main-es2015.bcd08324eaf7280a7df8.js:1:25459)
	next (main-es2015.bcd08324eaf7280a7df8.js:1:89060)
	(anonymous function) (main-es2015.bcd08324eaf7280a7df8.js:1:79728)
	__tryOrUnsub (main-es2015.bcd08324eaf7280a7df8.js:1:6984)
	next (main-es2015.bcd08324eaf7280a7df8.js:1:6213)
	_next (main-es2015.bcd08324eaf7280a7df8.js:1:5376)
	next (main-es2015.bcd08324eaf7280a7df8.js:1:5151)
	next (main-es2015.bcd08324eaf7280a7df8.js:1:202041)
	emit (main-es2015.bcd08324eaf7280a7df8.js:1:79599)
	run (polyfills-es5.5257e1a0419cc7d43246.js:1:4477)
	onHandleError (main-es2015.bcd08324eaf7280a7df8.js:1:83651)
	runGuarded (polyfills-es5.5257e1a0419cc7d43246.js:1:4695)
	t (polyfills-es5.5257e1a0419cc7d43246.js:1:14334)
	microtaskDrainDone (polyfills-es5.5257e1a0419cc7d43246.js:1:14383)
	g (polyfills-es5.5257e1a0419cc7d43246.js:1:12293)
	invokeTask (polyfills-es5.5257e1a0419cc7d43246.js:1:11065)
	b (polyfills-es5.5257e1a0419cc7d43246.js:1:24247)
	m (polyfills-es5.5257e1a0419cc7d43246.js:1:24472)
```

Both these errors appear only on Safari Mobile with iOS 10.x.x.

If I replace "target: es2015" with "target: es5", everything stats to work as expected.

**Comment by Ks89** [comment]
Since the discussion seems dead, I opened a new issue about this topic here https://github.com/angular/angular/issues/31816
